packages/-ember-data/tests/integration/identifiers/polymorphic-scenarios-test.ts

```diff
@@ -61,7 +61,31 @@ module('Integration | Identifiers - single-table-inheritance polymorphic scenari
       store = owner.lookup('service:store');
     });
 
-    test(`Identity of polymorphic relations can change type`, async function(assert) {
+    test(`Identity of polymorphic relations can change type on first load`, async function(assert) {
+      const { owner } = this;
+      class TestAdapter extends Adapter {
+        shouldBackgroundReloadRecord() {
+          return false;
+        }
+        findRecord(_, __, id) {
+          return resolve({
+            data: {
+              id,
+              type: 'ferrari',
+              attributes: {
+                color: 'red',
+              },
+            },
+          });
+        }
+      }
+      owner.register('adapter:application', TestAdapter);
+
+      const foundFerrari = await store.findRecord('car', '1');
+      assert.strictEqual(foundFerrari.constructor.modelName, 'ferrari', 'We found the right type');
```

Comment:
This looks great, thanks.

Do you mind adding a few more assertions

1. assert that `{ type: 'car', id }` is in the cache
2. assert that `{ type: 'ferrari', id }` is in the cache
3. assert that they point to the same record

Reply:
Hi @hjdivad, thanks for the feedback. I just added this to the test, I hope this is what you meant:

```js
const cachedFerrari = await store.peekRecord('ferrari', '1');
assert.strictEqual(cachedFerrari.constructor.modelName, 'ferrari', 'We cached the right type');
```

However I tried adding one for the `car` type of record and it's actually not in cache, `store.peekRecord('car', '1')` return null. Isn't this expected? 🤔

Reply:
@krasnoukhov depends on how the test sets up identifiers; it should be possible to configure it either way, but the utility of having both `car: 1` and `ferrari: 1` point to the same record in the identity map is that you could then later `store.findRecord('car', 1)` and get back the ferrari without "forgetting" that it had already been loaded

Reply:
Let me try this now!

Reply:
However I'm not sure still... I have `store.findRecord('car', '1')` statement in the test already, not sure if I see the point of running it again and then asserting the equality with previous run?

Reply:
@krasnoukhov `store.peekRecord` would be equivalent to doing another `store.findRecord` here, i.e. verify that it's in the cache.